modules/auxiliary/gather/nis_bootparamd_domain.rb

```diff
@@ -0,0 +1,151 @@
+##
+# This module requires Metasploit: https://metasploit.com/download
+# Current source: https://github.com/rapid7/metasploit-framework
+##
+
+class MetasploitModule < Msf::Auxiliary
+
+  include Msf::Exploit::Remote::SunRPC
+  include Msf::Auxiliary::Report
+
+  def initialize(info = {})
+    super(update_info(info,
+      'Name'        => 'NIS bootparamd Domain Name Disclosure',
+      'Description' => %q{
+        This module discloses the NIS domain name from bootparamd.
+
+        You must know a client address from the target's bootparams file.
+
+        Hint: try hosts within the same network range as the target.
+      },
+      'Author'      => [
+        'SATAN',         # boot.c
+        'pentestmonkey', # Blog post
+        'wvu'            # Metasploit module
+      ],
+      'References'  => [
+        ['URL', 'https://tools.ietf.org/html/rfc1831'],
+        ['URL', 'https://tools.ietf.org/html/rfc4506'],
+        ['URL', 'http://pentestmonkey.net/blog/nis-domain-name']
+      ],
+      'License'     => MSF_LICENSE
+    ))
+
+    register_options([
+      OptEnum.new('PROTOCOL',  [true, 'Protocol to use', 'udp', %w{tcp udp}]),
+      OptAddress.new('CLIENT', [true, 'Client from target\'s bootparams file'])
```

Comment:
Double quotes is preferred over escaping the single quote.

Reply:
As a former double-quote evangelist, I can get behind that.